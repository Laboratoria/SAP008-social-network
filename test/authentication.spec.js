import {
  googleLogin,
  login,
  register,
  signOut,
  getUserUid,
  getDisplayName,
  getPhotoUser,
  createCollection,
} from '../src/lib/authentication.js';

beforeEach(() => jest.clearAllMocks());

describe('google Login', () => {
  it('is a function', () => {
    expect(typeof googleLogin).toBe('function');
  });

  it('should call Firebase signInWithPopup function', () => {
    googleLogin('provider');
    expect(firebase.auth).toHaveBeenCalledTimes(1);
  });
});

describe('login', () => {
  it('is a function', () => {
    expect(typeof login).toBe('function');
  });

  it('should call Firebase signInWithEmailAndPassword function', () => {
    const email = 'testevanellen@hotmail.com';
    const password = '123456';
    login(email, password);
    expect(firebase.auth).toHaveBeenCalledTimes(1);
  });
});

describe('register', () => {
  it('is a function', () => {
    expect(typeof register).toBe('function');
  });

  it('should call Firebase createUserWithEmailAndPassword( function', () => {
    const email = 'testevanellen@hotmail.com';
    const password = '123456';
    const username = 'vanellen';
    register(email, password, username);
    expect(firebase.auth).toHaveBeenCalledTimes(1);
  });
});

describe('signOut', () => {
  it('should be a function', () => {
    expect(typeof signOut).toBe('function');
  });

  it('should call Firebase signOut function', () => {
    signOut();
    expect(firebase.auth).toHaveBeenCalledTimes(1);
  });
});

describe('UserUid', () => {
  it('is a function', () => {
    expect(typeof getUserUid).toBe('function');
  });

  it('should call Firebase getUserUid function', () => {
    getUserUid();
    expect(firebase.auth).toHaveBeenCalledTimes(1);
  });
});

describe('UserName', () => {
  it('is a function', () => {
    expect(typeof getDisplayName).toBe('function');
  });

  it('should call Firebase getDisplayName function', () => {
    getDisplayName();
    expect(firebase.auth).toHaveBeenCalledTimes(1);
  });
});

describe('UserPhoto', () => {
  it('is a function', () => {
    expect(typeof getPhotoUser).toBe('function');
  });

  it('should call Firebase getPhotoUser function', () => {
    getPhotoUser();
    expect(firebase.auth).toHaveBeenCalledTimes(1);
  });
});

describe('createCollection', () => {
  it('should be a function', () => {
    expect(typeof createCollection).toBe('function');
  });

  it('should call Firebase createCollection function', () => {
    createCollection();
    expect(firebase.firestore).toHaveBeenCalledTimes(1);
  });
});
