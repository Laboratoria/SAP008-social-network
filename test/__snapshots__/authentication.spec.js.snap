// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render página de login carrega a página 1`] = `undefined`;
